src/qdsht.jl

```diff
@@ -0,0 +1,175 @@
+
+"
+    mul!(Y, Q::QSDHT, A)
+
+Calculate the forward quasi-discrete spherical Hankel transform of array `A` using the
+QDSHT `Q` and store the result in `Y`.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(1e-2, 8); A = exp.(-q.r.^2/(1e-3*q.R)); Y = similar(A);
+julia> mul!(Y, q, A)
+8-element Array{Float64,1}:
+  8.73574297951054e-9
+  4.166950654425497e-9
+  1.2135269740167775e-9
+  2.1570485709886677e-10
+  2.3441730449103528e-11
+  1.5335062911366657e-12
+  7.37934569592307e-14
+ -3.450933124215762e-15
+```
+"
+mul!(::Any, ::QDSHT, ::Any)
+
+"
+    ldiv!(Y, Q::QDSHT, A)
+
+Calculate the inverse quasi-discrete spherical Hankel transform of array `A` using the QDSHT
+`Q` and store the result in `Y`.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(1e-2, 8); A = exp.(-q.r.^2/(1e-3*q.R)); Y = similar(A);
+julia> mul!(Y, q, A);
+julia> YY = similar(Y); ldiv!(YY, q, Y);
+julia> YY ≈ A
+true
+```
+"
+ldiv!(::Any, ::QDSHT, ::Any)
+
+"""
+    integrateR(A, Q::QDSHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in ``n+1``-dimensional real space.
+
+Assuming `A` contains samples of a function `f(r)` at sample points `Q.r`, then
+`integrateR(A, Q)` approximates ∫f(r)rⁿ dr from r=0 to r=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+!!! warning
+    using `integrateR` to integrate a function (i.e. `A` rather than `abs2(A)`) is only
+    supported for the 0th-order QDSHT. For more details see [Derivations](@ref).
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ √π/4 # analytical solution of ∫exp(-r²)rⁿ dr from 0 to ∞
+true
+```
+"""
+integrateR(::Any, ::QDSHT)
+
+"""
+    integrateK(Ak, Q::QDSHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in ``n+1``-dimensional reciprocal space.
+
+Assuming `A` contains samples of a function `f(k)` at sample points `Q.k`, then
+`integrateR(A, Q)` approximates ∫f(k)kⁿ dk from k=0 to k=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ √π/4 # analytical solution of ∫exp(-r²)rⁿ dr from 0 to ∞
+true
+julia> Ak = q*A;
+julia> integrateK(abs2.(Ak), q) ≈ √π/4 # Same result
+true
+```
+"""
+integrateK(::Any, ::QDSHT)
+
+"""
+    onaxis(Ak, Q::QDSHT; dim=Q.dim)
+
+Calculate on-axis sample in space (i.e. at r=0) from transformed array `Ak`.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> onaxis(q*A, q) ≈ 1 # should be exp(0) = 1
+true
+```
+"""
+function onaxis(Ak, Q::QDSHT; dim = Q.dim)
```

Comment:
Presumably for spherical transforms this gets the point at (0, 0, 0)? If so we should consider renaming it to `origin` instead to be more general

Reply:
Good point, yes, although since we're only looking at the 1D radial component, `onaxis` also makes sense.